From 12b89b523d13110c3b0d5adfb84b2faca76cff77 Mon Sep 17 00:00:00 2001
From: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
Date: Mon, 5 Jun 2017 18:55:18 -0500
Subject: [PATCH 1846/5242] MGS-2954 arm64: dts: mx8: GPU: Update Frequencies
 to match with design

commit  f70f1f93679547928665cf34f52fba72d9b50cec from
https://source.codeaurora.org/external/imx/linux-imx.git

Update GPU frequencies for 8QM and 8QXP to match with design.

Signed-off-by: Prabhu Sundararaj <prabhu.sundararaj@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi  |    4 ++--
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   11 ++++++++++-
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index c129734..59b0528 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -1353,7 +1353,7 @@
 		clocks = <&clk IMX8QM_GPU0_CORE_CLK>, <&clk IMX8QM_GPU0_SHADER_CLK>;
 		clock-names = "core", "shader";
 		assigned-clocks = <&clk IMX8QM_GPU0_CORE_CLK>, <&clk IMX8QM_GPU0_SHADER_CLK>;
-		assigned-clock-rates = <650000000>, <700000000>;
+		assigned-clock-rates = <800000000>, <1000000000>;
 		fsl,sc_gpu_pid = <SC_R_GPU_0_PID0>;
 		power-domains = <&pd_gpu0>;
 		status = "disabled";
@@ -1366,7 +1366,7 @@
 		clocks = <&clk IMX8QM_GPU1_CORE_CLK>, <&clk IMX8QM_GPU1_SHADER_CLK>;
 		clock-names = "core", "shader";
 		assigned-clocks = <&clk IMX8QM_GPU1_CORE_CLK>, <&clk IMX8QM_GPU1_SHADER_CLK>;
-		assigned-clock-rates = <650000000>, <700000000>;
+		assigned-clock-rates = <800000000>, <1000000000>;
 		fsl,sc_gpu_pid = <SC_R_GPU_1_PID0>;
 		power-domains = <&pd_gpu1>;
 		status = "disabled";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 401eb7f..06fe207 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -784,7 +784,16 @@
 		clocks = <&clk IMX8QXP_GPU0_CORE_CLK>, <&clk IMX8QXP_GPU0_SHADER_CLK>;
 		clock-names = "core", "shader";
 		assigned-clocks = <&clk IMX8QXP_GPU0_CORE_CLK>, <&clk IMX8QXP_GPU0_SHADER_CLK>;
-		assigned-clock-rates = <650000000>, <700000000>;
+		assigned-clock-rates = <600000000>, <850000000>;
+		power-domains = <&pd_gpu0>;
+		status = "disabled";
+	};
+
+	imx8_gpu_ss: imx8_gpu_ss {
+		compatible = "fsl,imx8qxp-gpu", "fsl,imx8-gpu-ss";
+		cores = <&gpu_3d0>;
+		reg = <0x0 0x80000000 0x0 0x80000000>, <0x0 0x0 0x0 0x8000000>;
+		reg-names = "phys_baseaddr", "contiguous_mem";
 		status = "disabled";
 	};
 
-- 
1.7.9.5

